### PR TITLE
feat(nfts): add in time based auctions

### PR DESCRIPTION
**Pull request by rsteubs-bc**
## Description
- Dutch Auction (descending)
- English Auction (ascending)

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts [hunk 1]
```diff
@@ -229,9 +229,9 @@ export default ({ api }: { api: APIType }) => {
       yield put(A.fetchFeesLoading())
       const signer: Signer = yield call(getEthSigner)
       let fees: GasDataI
+      // TODO: DONT DEFAULT TO 1 WEEK
+      const expirationTime = moment().add(14, 'day').unix()
       if (action.payload.operation === GasCalculationOperations.Buy) {
-        // TODO: DONT DEFAULT TO 1 WEEK
-        const expirationTime = moment().add(7, 'day').unix()
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts:234]: Ideally a user is able to select this via some widget.

File: packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts [hunk 2]
```diff
@@ -463,7 +473,7 @@ export default ({ api }: { api: APIType }) => {
       yield put(
         A.setMarketplaceData({
           // @ts-ignore
-          collection: IS_TESTNET ? { ...res, collection_data: { ...res } } : res
+          collection: res
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts:476]: I think this will break testnet but checking

Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts:476]: Yeah think we need this

Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts:476]: Not sure why the BE returns different shapes for this call, I think for testnet we reroute directly to openseas api via service-explorer-gateway and for mainnet hit our internal db

Comment (rsteubs-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts:476]: From pulling from our api locally on testnet, had to change this so collection_data would come through

File: packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts (hunk 1 above)
Comment (rsteubs-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts:234]: Yeah, working on that right now.  https://github.com/blockchain/blockchain-wallet-v4-frontend/pull/4080/commits/a75d88aa6d919c505e9a1f4e4e3b1bad5dc5a90c

File: packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts [hunk 3]
```diff
@@ -263,12 +263,25 @@ export default ({ api }: { api: APIType }) => {
         )
         yield put(A.fetchFeesSuccess(fees))
       } else if (action.payload.operation === GasCalculationOperations.Sell) {
+        const listingTime =
+          action.payload.listingTime !== '' && action.payload.listingTime !== undefined
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/nfts/sagas.ts:267]: super pedantic but this could be `!action.payload.listingTime`

**Comment by plondon** [comment]
A couple of UX improvements that would be nice (ordered by priority):
1. for sell with declining price do form validation that ending price is less than start price
2. default start date to current date/time
3.  if you open the calendar and don't select a date but click out of the pop up it persists
4. end date calendar pop out overflows the modal container
5. keep the label (Start Date, End Date) in the same row as the input fields

**Pull request merged**